postrelease-vip.php

```diff
@@ -2,567 +2,680 @@
+				}
+
+				// these functions require security check to be called
+				if ( $security_check_ok && $this->postrelease_authenticate_IP() ) {
+					// enable plugin during sign up process
+					if ( 'enable' === $function ) {
+						if ( isset( $_REQUEST['status'] ) ) {
+							$this->postrelease_enable_plugin( sanitize_text_field( $_REQUEST['status'] ) );
+						 // get latest posts so that our server can index blog's content
+						} else if ( 'getposts' === $function ) {
+							$this->postrelease_get_posts_xml_for_indexing();
+						} else if ( 'status' === $function ) {
+							$this->postrelease_get_status();
+						} else if ( 'check' === $function ) {
+							$this->postrelease_get_check();
+						}
+						exit;
+					}
+				}
+			}
+		}
 
-/**
- * Change author display name to "Promoted" in secondary impression
- */
-function postrelease_the_author($author) {
-	global $post;
-	$template_post_id = get_option('prx_template_post_id');
-	if(isset($post) && $post->ID == get_option('prx_template_post_id', -1)) {
-		$author = 'Promoted';
-	}
-	return $author;
-}
+		/**
+		 * Handle enable request/send success response
+		 *
+		 * @return void
+		 */
+		public function postrelease_handle_enable_request() {
+			if (
+				isset( $_GET['do'] )
+				&& 'update' == $_GET['do']
+				&& isset( $_GET['postrelease_enable'] )
+				&& '1' == $_GET['postrelease_enable']
+			) {
+				$this->postrelease_send_success_response();
+				exit;
+			}
+		}
 
-/**
- * Change author display name to "Promoted" in secondary impression 
- */
-function postrelease_the_author_display_name($author) {
-	global $post;
-	$template_post_id = get_option('prx_template_post_id');
-	if(isset($post) && $post->ID == get_option('prx_template_post_id', -1)) {
-		$author = 'Promoted';
-	}
-	return $author;
-}
+		/**
+		 * This code returns XML that follows the schema
+		 * specified in PostRelease API documentation.
+		 * This XML will be used to hydrate our indexing system that is necessary to
+		 * suggest what are the best publications for an ad.
+		 *
+		 * @return void, outputs XML
+		 */
+		public function postrelease_get_posts_xml_for_indexing() {
+			// by default return 200 posts
+			$default_number_posts = 200;
+
+			// sanitizing num URL parameter
+			$number_posts = isset( $_REQUEST['num'] ) ? intval( $_REQUEST['num'] ) : 0;
+
+			if ( 0 > $number_posts || 200 < $number_posts ) {
+				$number_posts = $default_number_posts;
+			}
 
-/**
- * Change author link to blog homepage URL in secondary impression
- */
-function postrelease_author_link($link, $author_id) {
-	global $post;
-	if(isset($post) && $post->ID == get_option('prx_template_post_id', -1)) {
-		$link = home_url();
-	}
-	return $link;
-}
+			$args = [
+				'posts_per_page' => $number_posts,
+				'ignore_sticky_posts' => true,
+				'no_found_rows' => true,
+			];
+			$posts_for_xml = new WP_Query( $args );
+
+			if ( $posts_for_xml->have_posts() ) {
+				$xml = '<?xml version="1.0" encoding="UTF-8" ?>';
```

Comment:
It's unnecessary to store all this data in a variable, then do nothing with it but echo. Output as you go.